Mathlib/LinearAlgebra/Matrix/GeneralLinearGroup.lean

```diff
@@ -329,4 +331,98 @@ def planeConformalMatrix {R} [Field R] (a b : R) (hab : a ^ 2 + b ^ 2 ≠ 0) :
 -/
 end Examples
 
+section cardinal
+
+variable (n : ℕ) {𝔽 : Type*} [Field 𝔽] [Fintype 𝔽]
+
+local notation "q" => Fintype.card 𝔽
+
+noncomputable instance {k : ℕ} :
+    Fintype ({ s : Fin k → (Fin n → 𝔽) // LinearIndependent 𝔽 s}) :=
+  Fintype.ofFinite _
+
+noncomputable instance {k : ℕ} {s : { s : Fin k → Fin n → 𝔽 // LinearIndependent 𝔽 s }} :
+    Fintype (Submodule.span 𝔽 (Set.range (s : Fin k → Fin n → 𝔽)) : Set (Fin n → 𝔽)) :=
+  Fintype.ofFinite _
+
+noncomputable instance {k : ℕ} {s : { s : Fin k → Fin n → 𝔽 // LinearIndependent 𝔽 s }} :
+    Fintype ((Submodule.span 𝔽 (Set.range (s : Fin k → Fin n → 𝔽)))ᶜ : Set (Fin n → 𝔽)) :=
+  Fintype.ofFinite _
+
+lemma complement_card {k : ℕ} (s : { s : Fin k → Fin n → 𝔽 // LinearIndependent 𝔽 s }):
+    Fintype.card ((Submodule.span 𝔽 (Set.range (s : Fin k → Fin n → 𝔽)))ᶜ : Set (Fin n → 𝔽)) =
+      (q) ^ n - (q) ^ k := by
+  rw [Fintype.card_compl_set, Fintype.card_fun, Fintype.card_fin]
+  simp only [SetLike.coe_sort_coe]
+  rw [card_eq_pow_finrank (K := 𝔽) (V := Submodule.span 𝔽 (Set.range (s : Fin k → Fin n → 𝔽))),
+    finrank_span_eq_card s.property, Fintype.card_fin]
+
+def inductiveStepEquiv {k : ℕ} :
+    { s : Fin (k + 1) → Fin n → 𝔽 // LinearIndependent 𝔽 s } ≃
+      Σ s : { s : Fin k → Fin n → 𝔽 // LinearIndependent 𝔽 s },
+        ((Submodule.span 𝔽 (Set.range (s : Fin k → Fin n → 𝔽)))ᶜ : Set (Fin n → 𝔽)) where
+  toFun s := by
+    have := linearIndependent_fin_succ.mp s.property
+    use ⟨Fin.tail s.val, this.left⟩
+    exact ⟨s.val 0, this.right⟩
+  invFun s := by
+    use Fin.cons s.2.val s.1.val
+    exact linearIndependent_fin_cons.mpr ⟨s.1.property, s.2.property⟩
+  left_inv _ := by simp only [Fin.cons_self_tail, Subtype.coe_eta]
+  right_inv := fun ⟨_, _⟩ => by simp only [Fin.cons_zero, Subtype.coe_eta, Sigma.mk.inj_iff,
+    Fin.tail_cons, heq_eq_eq, and_self]
+
```

Comment:
The bot is complaining about the lines being too long. There is a 100 character limit. If you put `#lint` at the end of the file it will tell you any things that need fixing, although my suggestion should fix it.